contrib/babelfishpg_tsql/sql/babelfishpg_tsql.sql

```diff
@@ -3760,23 +3760,28 @@ LANGUAGE pltsql
 AS $$
 DECLARE @input_loginname sys.sysname;
 DECLARE @current_username sys.nvarchar(128)
+DECLARE @current_userid Oid
 DECLARE @is_sysadmin BIT
+DECLARE @is_securityadmin BIT
 BEGIN
 
-    IF is_srvrolemember('securityadmin') = 0 
+    SET @current_userid = suser_id();
+    SET @is_sysadmin = pg_has_role(@current_userid, cast('sysadmin' as text), 'MEMBER');
+    SET @is_securityadmin = pg_has_role(@current_userid, cast('securityadmin' as text), 'MEMBER');
+
+    IF @is_securityadmin = 0 AND @is_sysadmin = 0 
     BEGIN
         RAISERROR('User does not have permission to perform this action.', 16, 1);
 		RETURN 0;
     END
 
     SET @current_username = LOWER(sys.suser_name());
```

Comment:
The value of `@current_username` for GAD purposes should be derived from session_user, which is an internal-specific change.
Please ensure that GAD-related scenarios are handled only within the internal environment and not in OSS.
This PR should include only the non-GAD session changes.

Reply:
If we can come up with something which works with both correctly will be great as well.